apis/pkg/v1beta1/revision_types.go

```diff
@@ -78,6 +79,13 @@ type PackageRevisionSpec struct {
 	// +optional
 	// +kubebuilder:default=false
 	SkipDependencyResolution *bool `json:"skipDependencyResolution,omitempty"`
+
+	// PermissionRequests for RBAC rules required for this provider's controller
+	// to function. The RBAC manager is responsible for granting requested
+	// permissions. It uses a whitelist to determine what permissions a provider
+	// may be granted.
+	// +optional
+	PermissionRequests []rbacv1.PolicyRule `json:"permissionRequests,omitempty"`
```

Comment:
Also see my comment below about potentially annotating / labelling the SA we create with additional permission requests.

Reply:
I didn't think much about this - the spec just seemed to be the intuitive place given that requesting permissions seems to be a form of expressing desired state.

Reply:
Yeah I see what you are saying. I think my rationale is that if something like a `Pod` has a container image in the spec, when it pulls the image and runs it the digest and other information about that image is put in the status because pulling that image and running it is a result of operations performed based on the desired spec. This feels kind of like that, the difference here is just that we have another part of the system explicitly depending on a status field which feels weird.

Reply:
To me the issue isn't so much whether another part of the system depends on this information, but rather whether we consider the permission requests to be an input to how the provider revision is reconciled (i.e. its desired state), or an output of how it was reconciled (i.e. observed state). This isn't so clear cut in this case.

My thinking was that the creation of Deployments, CRDs, RBAC roles, etc happen in response to the existence of the ProviderRevision. The ProviderRevision is thus responsible for declaring the desired state of those things and thus the permission requests fit in its spec.

However another way of looking at it - which I think probably fits better with how ProviderRevisions work today - is that a ProviderRevision is basically a pointer to something else (the provider package) that captures the desired state. So the relationship isn't `package tarball -> revision -> CRDs etc` but rather `revision -> package tarball -> CRDs etc`. Under this view I can see how the revisions desired state is "go unpack this package and do what it says", and thus "the package said to do this" is fair game as status.

TL;DR Putting this in status still feels kinda weird to me, but I think on reflection you're right that it does fit better with the current implementation to put it there. I forget - would we recover from lost status here? i.e. Would we unpack the package again and recreate the status if it were lost?

Reply:
@negz yep! we check the package contents on every reconcile because we must guarantee installed objects still match.

Reply:
Okay, I pushed a commit that moves these to the status.